packages/lib-core/src/index.ts

```diff
@@ -7,14 +6,12 @@ export { useExtensions } from './store/useExtensions';
 export { usePluginInfo } from './store/usePluginInfo';
 export { useResolvedExtensions } from './store/useResolvedExtensions';
 export {
+  CodeRef,
   Extension,
   ExtensionPredicate,
-  LoadedExtension,
   EncodedExtension,
+  LoadedExtension,
   ResolvedExtension,
 } from './types/extension';
 export { PluginEventType, PluginInfoEntry, PluginConsumer, PluginManager } from './types/store';
 export { PluginManifest, PluginRuntimeMetadata, LoadedPlugin } from './types/plugin';
-
-// Core extension types, to be exported via lib-extensions package
-export * from './extensions';
```

Comment:
@vojtechszocs we originally went through the ebb and flow of removing this `export *`, then adding it back. I believe @karelhala added this to prevent missing extensions... I want to confirm - since the only thing remaining is the core extensions, the concern of missing some is less prevalent (as lib-extensions has the `export *` now)? If so - I think this is ok.

Reply:
This "export all extension types" was just moved to `lib-extensions` package, becoming its main scope and purpose.

Since the scope of `lib-extensions` is very narrow, having `export *` in its main entry point is OK, i.e. we generally want to expose everything from that package to consumers.

However, the scope of `lib-core` is much broader - basically it should provide essential APIs to
- **[host apps]** set up runtime plugin infrastructure via `PluginStore` and `PluginStoreProvider`
- **[host apps and/or plugins]** allow consuming extensions via React hooks like `useExtensions`

I generally avoid `export *` in a library's entry point because it exposes everything down to all consumers. I think that API surface should be as small as possible, which (among other things) minimizes the amount of crazy/unexpected ways people use the library which then sparks more issues and maintenance efforts.

Having full control over what we export (via named exports) is basically a way to minimize the API surface. I agree that it's less convenient, but I think it's better in the long term.